server/src/main/kotlin/core/gameflow/HandState.kt

```diff
@@ -3,32 +3,54 @@ package core.gameflow
 import core.Card
 
 /**
- * activePlayer set to null indicates that there is no further action possible in given betting round
+ * activePlayer set to null indicates that there is no further action possible in given betting round.
  *
  * lastAggressor parameter should be used to determine when the action in a betting round should end,
- * as well as to resolve the order in which players show their cards during the showdown. That's why
- * every time a betting rounds starts with a check, the first player that checks should be marked
- * as the last aggressor (even though no raise was made). The same rule applies when UTG calls the big blind
- * in pre-flop betting round.
+ * as well as to resolve the order in which players show their cards during the showdown. If set to null,
+ * it indicates that no aggressive action was taken yet.
+ *
+ * extraBet is used when there is an incomplete raise all-in. It is used to store the additional bet value.
+ * It should be reset to 0 every time a legal bet / raise is made.
  */
 data class HandState(
     val players: List<Player>,
     val blinds: Blinds,
     val buttonPosition: Int,
     val activePlayer: Player?,
     val communityCards: List<Card> = emptyList(),
+    val bettingRound: BettingRound = BettingRound.PRE_FLOP,
     val lastAggressor: Player? = null,
-    val currentBet: Int = 0,
-    val minRaise: Int = blinds.bigBlind
+    val lastLegalBet: Int = 0,
+    val minRaise: Int = blinds.bigBlind,
```

Comment:
shouldnt the default (1st round) min raise/bet be bigBlind * 2?

Reply:
That's true for pre-flop, false for any other betting round, so we can do it either way.

Reply:
Lets leave it this way I guess